locator/abstract_replication_strategy.hh

```diff
@@ -207,6 +207,9 @@ public:
     /// new set of replicas differs from the old one.
     virtual std::optional<inet_address_vector_replica_set> get_endpoints_for_reading(const token& search_token) const = 0;
 
+    /// Returns true if there are any pending ranges for this endpoint.
+    virtual bool has_pending_ranges(inet_address endpoint) const = 0;
```

Comment:
I'd leave a comment that this is very expensive.

Reply:
done